mllib/src/test/scala/org/apache/spark/ml/feature/VectorIndexerSuite.scala

```diff
@@ -219,6 +231,33 @@ class VectorIndexerSuite extends SparkFunSuite with MLlibTestSparkContext
     checkCategoryMaps(densePoints2, maxCategories = 2, categoricalFeatures = Set(1, 3))
   }
 
+  test("handle invalid") {
+    for ((points, pointsTestInvalid) <- Seq((densePoints1, densePoints1TestInvalid),
+      (sparsePoints1, sparsePoints1TestInvalid))) {
+      intercept[SparkException] {
```

Comment:
Only wrap the part which should throw the exception in the intercept.  Otherwise, it might be masking some other error.